packages/fungible-token/bridge-fungible-token/tests/harness.rs

```diff
@@ -266,6 +267,141 @@ mod success {
         assert_eq!(amount, config.overflow_2);
     }
 
+    #[tokio::test]
+    async fn claim_refund_of_wrong_token_deposit() {
+        // Send a message informing about a deposit with a a random token address, different from the bridged token
+        // Upon sending this message, the contract will register a refund for the deposit and random token
+        // - Verify that the contract state has correctly changed: new refund record inserted for the correct amount and the random token
+        // - Verify that the contract emits the correct logs
+        // - Verify that the the receipt of the transaction contains a message for the L1 Portal that allows to withdraw the above mentioned deposit
+        let mut wallet = env::setup_wallet();
+
+        let configurables: Option<BridgeFungibleTokenContractConfigurables> = None;
+        let config = env::generate_test_config((BRIDGED_TOKEN_DECIMALS, PROXY_TOKEN_DECIMALS));
+        let wrong_token_value: &str =
+            "0x1111110000000000000000000000000000000000000000000000000000111111";
+
+        let (message, coin, deposit_contract) = env::construct_msg_data(
+            wrong_token_value,
+            FROM,
+            *wallet.address().hash(),
+            config.overflow_2,
+            configurables.clone(),
+            false,
+            None,
+        )
+        .await;
+
+        // Set up the environment
+        let (
+            test_contract,
+            contract_inputs,
+            coin_inputs,
+            message_inputs,
+            test_contract_id,
+            provider,
+        ) = env::setup_environment(
+            &mut wallet,
+            vec![coin],
+            vec![message],
+            deposit_contract,
+            None,
+            configurables,
+        )
+        .await;
+
+        // Relay the test message to the test contract
+        let receipts = env::relay_message_to_contract(
+            &wallet,
+            message_inputs[0].clone(),
+            contract_inputs,
+            &coin_inputs[..],
+            &env::generate_variable_output(),
+        )
+        .await;
+
+        let log_decoder = test_contract.log_decoder();
+        let refund_registered_event = log_decoder
+            .decode_logs_with_type::<RefundRegisteredEvent>(&receipts)
+            .unwrap();
+
+        // Verify the message value was received by the test contract
+        let test_contract_balance = provider
```

Comment:
I'm wondering what the 2nd arg here does - the default assetid. Could you remind me? Should we be using some specific asset instead?

Reply:
It sounds to me like it is the native token of the L2. You can see it in the way that messages are built in `environment.rs`: 

```rs
pub async fn construct_msg_data(
    token: &str,
    from: &str,
    to: [u8; 32],
    amount: Unsigned256,
    config: Option<BridgeFungibleTokenContractConfigurables>,
    deposit_to_contract: bool,
    // TODO: https://github.com/FuelLabs/bridge-fungible-token/issues/61
    extra_data: Option<Vec<u8>>,
) -> ((u64, Vec<u8>), (u64, AssetId), Option<ContractId>) {
    let mut message_data = Vec::with_capacity(5);
    message_data.append(&mut decode_hex(token));
    message_data.append(&mut decode_hex(from));
    message_data.append(&mut to.to_vec());
    message_data.append(&mut encode_hex(amount).to_vec());

    let mut deposit_recipient: Option<ContractId> = None;

    if deposit_to_contract {
        let hash = keccak_hash("DEPOSIT_TO_CONTRACT");
        let mut byte: Vec<u8> = vec![0u8];
        byte.copy_from_slice(&hash[..1]);
        message_data.append(&mut byte);
        deposit_recipient = Option::Some(ContractId::new(to));
    };

    if let Some(mut data) = extra_data {
        message_data.append(&mut data);
    };

    let message_data = prefix_contract_id(message_data, config).await;
    let message = (100, message_data);
    let coin = (DEFAULT_COIN_AMOUNT, AssetId::default());

    (message, coin, deposit_recipient)
}
```

There are two fungible amounts in this fn: one is the `amount`, which refers to the bridged asset, and the other is `coin`, which refers to the native asset.